### PR TITLE
[IMP][14.0] Module: pattern_import_export add domain for sub patterns

### PR DESCRIPTION
**Pull request by Kev-Roche**
Add domain on sub patterns
replace : https://github.com/shopinvader/pattern-import-export/pull/83

### CODE REVIEW

File: pattern_import_export/models/ir_exports_line.py [hunk 1]
```diff
@@ -137,9 +137,10 @@ def _check_required_fields(self):
     @api.depends("name")
     def _compute_related_level_field(self):
         for export_line in self:
-            if export_line.name and export_line.export_id.model_id.model:
+            # if export_line.name and export_line.export_id.model_id.model:
```
Comment (florian-dacosta) [inline pattern_import_export/models/ir_exports_line.py]: You can totally remove this line now, not need to keep some old not working lines even comented

File: pattern_import_export/views/pattern_config.xml [hunk 2]
```diff
@@ -153,12 +153,14 @@
...
                     'invisible': [('hidden_fields', 'ilike', 'number_occurence')]}"
                 />
+                <field name="related_model_name" invisible="1" />
                 <field
                     name="sub_pattern_config_id"
                     attrs="{
                     'required': [('required_fields', 'ilike', 'sub_pattern_config_id')],
                     'readonly': [('hidden_fields', 'ilike', 'sub_pattern_config_id')],
                     'invisible': [('hidden_fields', 'ilike', 'sub_pattern_config_id')]}"
+                    domain="[('model_id', '=', related_model_name)]"
```
Comment (florian-dacosta) [inline pattern_import_export/views/pattern_config.xml]: This is not related_model_name here but related_model_id

File: pattern_import_export/views/pattern_config.xml [hunk 3]
```diff
@@ -169,7 +171,6 @@
                        'readonly': [('hidden_fields', 'ilike', 'add_select_tab')]
                        }"
                 />
-                <field name="related_model_name" invisible="1" />
```
Comment (florian-dacosta) [inline pattern_import_export/views/pattern_config.xml]: no need to deplace the field, it make a useless diff

**Review by florian-dacosta — CHANGES_REQUESTED** [review]

**Comment by Kev-Roche** [comment]
@florian-dacosta , my bad, I fix it

**Review by florian-dacosta — APPROVED** [review]

**Review by kevinkhao — APPROVED** [review]
functional tests
LGTM

**Review by kevinkhao — CHANGES_REQUESTED** [review]
edit: my bad, wrong manipulation on my side. LGTM

**Review by kevinkhao — APPROVED** [review]

**Comment by bealdav** [comment]
Finally many unmerged PR by there. Which are the PSC ? @sebastienbeau

**Pull request merged**